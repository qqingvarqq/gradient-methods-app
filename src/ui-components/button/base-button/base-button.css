.base-button {
  height: auto;
  width: auto;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
}
