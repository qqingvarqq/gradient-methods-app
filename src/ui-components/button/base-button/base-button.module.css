.root {
  height: auto;
  width: auto;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  padding: 0;
  background: none;
}
.root:focus {
  outline: none;
}
