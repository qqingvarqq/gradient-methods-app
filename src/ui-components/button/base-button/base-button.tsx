import * as React from 'react';
import styles from './base-button.module.css';

export interface BaseButtonProps
  extends React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  > {}
export const BaseButton: React.FC<BaseButtonProps> = props => {
  const {className, ...rest} = props;
  return <button className={`${styles.root} ${className}`} {...rest}></button>;
};
