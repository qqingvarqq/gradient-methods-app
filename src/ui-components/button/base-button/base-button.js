import React from 'react';
import './base-button.css';

export default function BaseButton(props) {
  const {className, children, ...rest} = props;
  const classNames = 'base-button ' + (className || '');
  return (
    <div role="button" className={classNames} {...rest}>
      {children}
    </div>
  );
}
