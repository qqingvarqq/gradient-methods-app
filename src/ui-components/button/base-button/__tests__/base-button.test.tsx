import * as React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {BaseButton} from '../base-button';

describe('BaseButton', () => {
  it('should render without crashing', () => {
    const {asFragment} = render(<BaseButton>btn</BaseButton>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should be clickable', () => {
    const onClick = jest.fn();
    const {container} = render(<BaseButton onClick={onClick} />);
    // @ts-ignore
    fireEvent.click(container.firstChild);
    expect(onClick).toBeCalled();
  });
});
