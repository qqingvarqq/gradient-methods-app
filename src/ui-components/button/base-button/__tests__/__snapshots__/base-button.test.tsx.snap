// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton should render without crashing 1`] = `
<DocumentFragment>
  <button
    class="root "
  >
    btn
  </button>
</DocumentFragment>
`;
